djangocms_navigation/cms_config.py

```diff
@@ -1,71 +1,73 @@
+from functools import partial
+
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 
 from cms.app_base import CMSAppConfig, CMSAppExtension
 from cms.models import Page
 
-from .utils import purge_menu_cache, get_model
-from .constants import MENU_MODEL, ITEM_MODEL
-
-
-MenuContent = get_model(MENU_MODEL)
-MenuItem = get_model(ITEM_MODEL)
+from .utils import purge_menu_cache
+from .models import MenuContent
+from .models import MenuItem
 
 
 class NavigationCMSExtension(CMSAppExtension):
     def __init__(self):
         self.navigation_apps_models = {}
 
     def configure_app(self, cms_config):
-        if hasattr(cms_config, "navigation_models"):
-            navigation_app_models = getattr(cms_config, "navigation_models")
-            if isinstance(navigation_app_models, dict):
-                self.navigation_apps_models.update(navigation_app_models)
-            else:
-                raise ImproperlyConfigured(
-                    "navigation configuration must be a dictionary object"
-                )
-        else:
+        if not hasattr(cms_config, "navigation_models"):
             raise ImproperlyConfigured(
                 "cms_config.py must have navigation_models attribute"
             )
 
+        navigation_app_models = getattr(cms_config, "navigation_models")
+        if isinstance(navigation_app_models, dict):
+            self.navigation_apps_models.update(navigation_app_models)
+        else:
+            raise ImproperlyConfigured(
+                "navigation configuration must be a dictionary object"
+            )
+
 
-def copy_menu_content(original_content):
+def copy_menu_content(menu_model, item_model, original_content):
```

Comment:
We don't know if this could have been reused by other apps, because of this I think we should make the additional attributes optional: `def copy_menu_content(original_content, menu_model=MenuContent, item_model=MenuItem):`. Partial won't work in this scenario and we would have to wrap the function. Any ideas of how we could do that?

The solution designed here will be used as an example for any addons that reuse copy methods. I think it's important that we make them backwards compatible.

Reply:
You are right that it's better to make this backwards compatible. Making the signature the above looks to be the correct choice. 

However I think partial still does work. Take a look at the python docs: https://docs.python.org/3.6/library/functools.html#functools.partial

It should work with keyword arguments as well.
```
>>> from functools import partial
>>> basetwo = partial(int, base=2)
>>> basetwo.__doc__ = 'Convert base 2 string to an int.'
>>> basetwo('10010')
```

Reply:
You are right that it's better to make this backwards compatible. Making the signature the above looks to be the correct choice. 

However I think partial still does work. Take a look at the python docs: https://docs.python.org/3.6/library/functools.html#functools.partial

It should work with keyword arguments as well.
```
>>> from functools import partial
>>> basetwo = partial(int, base=2)
>>> basetwo.__doc__ = 'Convert base 2 string to an int.'
>>> basetwo('10010')
```

Reply:
Looks like I was wrong. How about we solve it this way?

```
def _copy_menu_content(original_content, menu_model, item_model):
   # snip 
    return new_content


complete_copy_menu_content = partial(
    _copy_menu_content,
    MenuContent,
    MenuItem
)


def copy_menu_content(original_content):
    """Copy the MenuContent object and deepcopy its menu items."""
    return complete_copy_menu_content(original_content)
```

Reply:
That way `copy_menu_content` keeps its original signature.